match_p.h

```diff
@@ -1,179 +1,44 @@
-/*
- * Set match_start to the longest match starting at the given string and
- * return its length. Matches shorter or equal to prev_length are discarded,
- * in which case the result is equal to prev_length and match_start is garbage.
- *
- * IN assertions: cur_match is the head of the hash chain for the current
- * string (strstart) and its distance is <= MAX_DIST, and prev_length >=1
- * OUT assertion: the match length is not greater than s->lookahead
- */
 
 #include "zbuild.h"
 #include "deflate.h"
+#include "functable.h"
 
-#if (defined(UNALIGNED_OK) && MAX_MATCH == 258)
-
-   /* ARM 32-bit clang/gcc builds perform better, on average, with std2. Both gcc and clang and define __GNUC__. */
-#  if defined(__GNUC__) && defined(__arm__) && !defined(__aarch64__)
-#    define std2_longest_match
-   /* Only use std3_longest_match for little_endian systems, also avoid using it with
-      non-gcc compilers since the __builtin_ctzl() function might not be optimized. */
-#  elif(defined(__GNUC__) && defined(HAVE_BUILTIN_CTZL) && ((__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__) \
-        || defined(__LITTLE_ENDIAN__)))
-#    define std3_longest_match
-#  elif(defined(_MSC_VER) && defined(_WIN32))
-#    define std3_longest_match
-#  else
-#    define std2_longest_match
-#  endif
+#ifndef BESTCMP_TYPE
+#define BESTCMP_TYPE
 
+#ifdef UNALIGNED_OK
+#if MIN_MATCH >= 4
+typedef uint32_t        bestcmp_t;
+#elif MIN_MATCH > 1
```

Comment:
`#elif MIN_MATCH > 1` could be `#elif MIN_MATCH >= 2` to match the leading if style

Reply:
Done.